cfme/tests/cli/test_appliance_update.py

```diff
@@ -88,3 +90,49 @@ def test_update_yum(appliance_preupdate, appliance):
     appliance_preupdate.evmserverd.start()
     appliance_preupdate.wait_for_web_ui()
     assert appliance.version == appliance_preupdate.version
+
+
+@pytest.fixture(scope="function")
+def enabled_embedded_appliance(appliance_preupdate):
+    """Takes a preconfigured appliance and enables the embedded ansible role"""
+    appliance_preupdate.enable_embedded_ansible_role()
+    assert appliance_preupdate.is_embedded_ansible_running
+    return appliance_preupdate
+
+
+@pytest.mark.uncollectif(lambda: not store.current_appliance.is_downstream)
+@pytest.mark.uncollectif(lambda: store.current_appliance.version < "5.8")
+def test_embedded_ansible_update(enabled_embedded_appliance, appliance, old_version):
+    """ Tests updating an appliance which has embedded ansible role enabled, also confirms that the
+        role continues to function correctly after the update has completed"""
+    set_default_repo = True
+    with enabled_embedded_appliance:
+        red_hat_updates = RedHatUpdates(
+            service='rhsm',
+            url=conf.cfme_data['redhat_updates']['registration']['rhsm']['url'],
+            username=conf.credentials['rhsm']['username'],
+            password=conf.credentials['rhsm']['password'],
+            set_default_repository=set_default_repo
+        )
+        red_hat_updates.update_registration(validate=False)
+        red_hat_updates.check_updates()
+        wait_for(
+            func=red_hat_updates.checked_updates,
+            func_args=[appliance.server.name],
+            delay=10,
+            num_sec=100,
+            fail_func=red_hat_updates.refresh
+        )
+        if red_hat_updates.platform_updates_available():
+            red_hat_updates.update_appliances()
```

Comment:
**REQUIRED**
This context block could go into separate fixture.

Reply:
@quarckster @lcouzens 1LP+RFR, looks like this comment hasn't been resolved though?